<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- param -->
  <arg name="param_file" default="$(find-pkg-share gmps_driver)/config/gmsp_driver.param.yaml"/>

  <!-- topic name -->
  <arg name="in_velocity_name" default="/velocity"/>
  <arg name="in_soft_reset_name" default="/gmps_soft_reset"/>
  <arg name="in_gmps_can_frame_name" default="/from_can_bus"/>
  <arg name="out_gmps_can_frame_name" default="/to_can_bus"/>
  <arg name="out_gmps_detect_name" default="/gmps_detect"/>
  <arg name="out_gmps_error_name" default="/gmps_error"/>

  <node pkg="gmps_driver" exec="gmps_driver" name="gmps_driver" output="log" >
    <param from="$(var param_file)"/>

    <remap from="in_velocity" to="$(var in_velocity_name)"/>
    <remap from="in_soft_reset" to="$(var in_soft_reset_name)"/>
    <remap from="in_gmps_can_frame" to="$(var in_gmps_can_frame_name)"/>
    <remap from="out_gmps_can_frame" to="$(var out_gmps_can_frame_name)"/>
    <remap from="out_gmps_detect" to="$(var out_gmps_detect_name)"/>
    <remap from="out_gmps_error" to="$(var out_gmps_error_name)"/>
  </node>
</launch>
